lib/api/decorators/property_schema_representer.rb

```diff
@@ -33,11 +33,11 @@
 module API
   module Decorators
     class PropertySchemaRepresenter < ::API::Decorators::Single
-      def initialize(type:, name:, current_user: nil)
+      def initialize(type:, name:, required: true, writable: true, current_user: nil)
         @type = type
         @name = name
-        @required = true
-        @writable = true
+        @required = required
+        @writable = writable
```

Comment:
You might want to consider adding ``@regularExpression`` for custom field validation.